var/spack/repos/builtin/packages/python/package.py

```diff
@@ -114,15 +119,17 @@ def install(self, spec, prefix):
                       'user configurations are present.').format(self.version))
 
         # Need this to allow python build to find the Python installation.
-        env['PYTHONHOME'], env['PYTHONPATH'] = prefix, prefix
-        env['MACOSX_DEPLOYMENT_TARGET'] = '10.6'
+        spack_env.set('PYTHONHOME', prefix)
+        spack_env.set('PYTHONPATH', prefix)
+        spack_env.set('MACOSX_DEPLOYMENT_TARGET', '10.6')
```

Comment:
There is a way to get this number for a specific platform since 2.3, apparently:

```
from platform import mac_ver
mac_ver()[0]
```

It seems to return an empty string on other platforms, so it should be safe to use.